### PR TITLE
[Mobile Payments] Allow card reader modals to use the tab bar controller for presentation

### LINKED ISSUE #4175: [Mobile Payments] Card Reader Settings V2 overlay borders visible during presentation
I was trying out the new card reader settings and I noticed the transparent dark overlay that's added behind the modal visibly slides from the bottom instead of fading in.

Recorded an example on the simulator with Slow Animations enabled:

modal-overlay-visible

### PR DESCRIPTION
**Pull request by allendav**
Fixes #4175

Changes
- Adds `CardPresentPaymentsModalViewController` to accepted presented controllers for MainTabBarController presentationController
- Removes the background for the card reader modal in the XIB

To Test
- Enter card reader settings, begin the reader search flow, observe no sliding background
- Enter order details, begin the reader search flow, observe no sliding background

Update release notes:
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/CardPresentPayments/CardPresentPaymentsModalViewController.xib [hunk 1]
```diff
@@ -2,7 +2,6 @@
 <document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="17701" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
     <device id="retina6_1" orientation="portrait" appearance="light"/>
     <dependencies>
-        <deployment identifier="iOS"/>
```
Comment (allendav) [inline WooCommerce/Classes/ViewRelated/CardPresentPayments/CardPresentPaymentsModalViewController.xib:5]: I am not sure why this change was introduced, or if it has any impact

**Review by ctarda — APPROVED** [review]
https://user-images.githubusercontent.com/2722505/118857570-3056aa00-b8a6-11eb-9270-fd7e85a30b59.mov

:shipit: 

And 10_000 extra points for the name of the branch. 💯
